pkg/sidecar/configmap/filesystem/filesystem_watcher_test.go

```diff
@@ -148,8 +148,9 @@ func TestReadConfigMap(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			var dir string
 			if tc.createDir {
-				dir = createTempDir(t)
-				defer os.RemoveAll(dir)
+				var cleanup func()
```

Comment:
just curious why you declare the var here but in the other uses you don't?

Reply:
`var dir string` is declared three lines above, as `dir` is set by either the 'then' or 'else' blocks. So I don't want to use `:=`, which would shadow `dir`. Therefore I need to use `=`, so `cleanup` needs to be defined.